tests/integration/test_cli.py

```diff
@@ -0,0 +1,38 @@
+import os
+import sys
+import subprocess
+
+import pytest
+from chalice.utils import OSUtils
+
+CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+PROJECT_DIR = os.path.join(CURRENT_DIR, 'testapp')
+ROOT_DIR = os.path.dirname(os.path.dirname(CURRENT_DIR))
+MAIN_FILE = os.path.join(ROOT_DIR, 'chalice', 'cli', '__init__.py')
+
+
+@pytest.fixture
+def local_app(tmpdir):
+    temp_dir_path = str(tmpdir)
+    OSUtils().copytree(PROJECT_DIR, temp_dir_path)
+    os.chdir(temp_dir_path)
+    sys.path.insert(0, ROOT_DIR)
+    return temp_dir_path
+
+
+def test_stack_trace_printed_on_error(local_app):
+    app_file = os.path.join(local_app, 'app.py')
+    with open(app_file, 'w') as f:
+        f.write(
+            'from chalice import Chalice\n'
+            'app = Chalice(app_name="test")\n'
+            'foobarbaz\n'
+        )
+    p = subprocess.Popen([sys.executable, MAIN_FILE, 'local'],
+                         stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    _, err = p.communicate()
+    rc = p.returncode
+
+    assert rc == 2
+    assert 'Traceback' in err.decode()
```

Comment:
We generally don't use `.decode()` calls without an encoding specified.  You can also just decode it once from the `communicate()` call above:
```
stderr = p.communicate()[1].decode('ascii')
```

Reply:
Fixed.